README.md

```diff
@@ -1,50 +1,79 @@
+## Prerequisites
 
-### Usage
+In order to be able to use this Action in the CI/CD pipeline of your service, there are two prerequisites:
 
-To use this Action, you will need:
+1. Your repository must have an associated NPM package registry, and you need a read/write registry token that you can pass into the Action.
+2. The Action must be able to access, online, the OpenAPI v3 spec for your service. (At Birdie, we use an earlier step of our CI/CD pipeline to deploy the relevant service, which exposes an endpoint to fetch the latest version of our OpenAPI spec. That means that when the Hatch action runs, it can access the latest version.)
 
-- An OpenAPI v3 spec of your service, which is accessible to the Action. Easiest might be to expose an endpoint that serves your service's OpenAPI spec
-- A GitHub NPM registry and an access token to publish packages to it
+## How to use
 
-You can include this Action in your GitHub workflow as follows:
+
+Once you have the prerequisites in place, you can include this Action in your GitHub workflow. The Action requires 4 parameters to be passed in:
+
+| Name                     | Description                                                                                      |
+| -----------              | -----------                                                                                      |
+| service_name             | The name of the service for which you wish to generate a client                                  |
+| openapi_spec_url         | The URL pointing at the service's OpenAPI v3 spec in JSON format                                 |
+| registry_namespace       | The name of the GitHub package registry namespace (without the @) of which your service is part. |
+| registry_token           | A read/write access token for the package registry, used to publish the generated client.        |
+
+See below for an example configuration:
 
 ```yaml
 generate-api-client:
   name: Generate API client
   needs: <any prerequisite earlier job>
   runs-on: ubuntu-latest
   steps:
-    - uses: actions/checkout@v2 # <-- this one is needed for hatch to access source code
+    - uses: actions/checkout@v2
     - name: Run Hatch action
-      uses: birdiecare/hatch@v0.0.12 # <-- use current version
+      uses: birdiecare/hatch@v0.0.12 
       with:
-        path: <e.g. https://staging.myservice.com/docs.json>
-        name: <Name of your service, e.g. my-service> # the generated client will automatically be called my-service-client
-        token: <your GitHub Registry Token>
+        service_name: my-service # the generated client will be named "my-service-client"
+        openapi_spec_url: https://my-service.io/docs.json
+        registry_namespace: mycompany # the package will be published as @mycompany/my-service-client
```

Comment:
This is optional in the NPM world. Can't we simply expect the `service_name` to be `@mycompany/my-service-client`? What we'd need to do though is to extract the namespace if it exists for the authentication bits, which doesn't sound too much and would simplify both other parts of the code and the configuration 🤔

Reply:
Yes, that's a good point, we can probably use a regex of some sort to extract the namespace for authentication. It actually makes sense to just have a `generated_client_name` property instead of building a client name based on the `service_name` and `-client`. It's better for the user to decide the full client name. (I've learnt my lesson from auto-generated replication slot names 😀 )

Reply:
See discussion below; GitHub packages _requires_ a scoped package.